File: app/src/main/java/hm/orz/chaos114/android/slideviewer/pref/SettingPrefsSchema.java
package hm.orz.chaos114.android.slideviewer.pref;

import com.rejasupotaro.android.kvs.annotations.Key;
import com.rejasupotaro.android.kvs.annotations.Table;

/**
 * Manage SharedPreferences data for SettingActivity.
 */
@Table(name = "setting")
public interface SettingPrefsSchema {
    @Key(name = "enable_ocr")
    boolean enableOcr = false;
}

File: modules/infra/src/main/java/hm/orz/chaos114/android/slideviewer/infra/pref/SettingPrefsSchema.kt
package hm.orz.chaos114.android.slideviewer.infra.pref

import com.rejasupotaro.android.kvs.annotations.Key
import com.rejasupotaro.android.kvs.annotations.Table

/**
 * Manage SharedPreferences data for SettingActivity.
 */
@Table(name = "setting")
interface SettingPrefsSchema {
    companion object {
        @Key(name = "enable_ocr")
        val enableOcr = false

        @Key(name = "selected_language")
        val selectedLanguage: String? = null
    }
}
